pkg/storage/s3/s3_test.go

```diff
@@ -0,0 +1,81 @@
+package s3
+
+import (
+	"context"
+	"testing"
+
+	logrTesting "github.com/go-logr/logr/testing"
+
+	velerov1alpha2 "github.com/openshift/managed-velero-operator/pkg/apis/managed/v1alpha2"
+)
+
+func TestSetInstanceBucketName(t *testing.T) {
+	tests := []struct {
+		name            string
+		awsClient       *mockAWSClient
+		bucketName      string
+		matchBucketName bool
+	}{
+		{
+			name:            "set new bucket name in instance status",
+			awsClient:       fakeEmptyClient,
+			bucketName:      "",
+			matchBucketName: false,
```

Comment:
this doesn't really check that it's set, it only checks that it's not a blank string. should we check that it `strings.HasPrefix(storageConstants.StorageBucketPrefix)`?

Reply:
I would assert that if `instance.Status.StorageBucket.Name` isn't an empty string then it is actually set.

That said however, your suggestion would definitely test the desired behavior more tightly.

Reply:
Okay, yes, that's true. I guess where it makes a difference is in the other test cases.. are we *really* setting it to what we intend.